### DISCUSSION TITLE
EventSource is not defined in Node environment

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#530] [Q&A] [answered]
I'm trying to use Realtime from a node environment, but EventSource is not defined. The polyfill https://www.npmjs.com/package/event-source-polyfill isn't working either.

Any thoughts on running in a nodejs environment?

**Comment by ganigeorgiev** [comment, marked as answer]
Nodejs does't have native EventSource implementation (yet). I don't recommend using this polyfill (see https://github.com/Yaffle/EventSource/issues/202).

I recommend https://github.com/EventSource/eventsource. To load it, you have to define it as global:

```
// npm install eventsource --save
global.EventSource = require('eventsource');
```

This and other nodejs specifics are mentioned in the Installation section of the JS SDK readme - https://github.com/pocketbase/js-sdk#installation

**Comment by RobinTail** [comment]
```ts
import { EventSource } from "undici";
```
